extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/JPAResourceReferenceProvider.java

```diff
@@ -56,8 +58,39 @@ public void destroy() {
                     };
                 }
             }
-        }
+        } else if (StatelessSession.class.equals(type)) {
+            PersistenceContext pc = getAnnotation(annotations, PersistenceContext.class);
+            if (pc != null) {
+                if (jpaConfig.isJtaEnabled()) {
+                    TransactionEntityManagers transactionEntityManagers = Arc.container()
+                            .instance(TransactionEntityManagers.class).get();
+                    ForwardingEntityManager entityManager = new ForwardingEntityManager() {
 
+                        @Override
+                        protected EntityManager delegate() {
+                            return transactionEntityManagers.getEntityManager(pc.unitName());
+                        }
+                    };
+                    return () -> entityManager;
```

Comment:
Looks like something fishy here, shouldn't you return `unwrap(SessionFactory.class).openStatelessSession()`. in this case? I fail to see how the right object is injected otherwise.

CC @mkouba

Reply:
So this should be used for `@PersistenceContext(unitName="foo") StatelessSession` injection, right? But we don't support multiple PUs yet, correct?

If possible, I'd remove this code completely because `ResourceReferenceProvider` should only be used for non-CDI EE resource injection. In this case, it would be probably better to use a custom qualifier or something like that...



. So I would remove this code completely and only support `@Inject StatelessSession` backed by the `DefaultStatelessSessionProducer`.  

@maxandersen  What the use case  for `@PersistenceContext StatelessSession`?